src/depot_gui.cpp

```diff
@@ -715,6 +719,16 @@ struct DepotWindow : Window {
 			this->generate_list = false;
 			DepotSortList(&this->vehicle_list);
 
+			this->check_unitnumber_digits = true;
+		}
+	}
+
+	void OnPaint() override
+	{
+		this->RefreshVehicleList();
+
+		if (this->check_unitnumber_digits) {
+			this->check_unitnumber_digits = false;
```

Comment:
If another call to `RefreshVehicleList()` happens right after this line, what is trying to be prevented it void.

This looks very much like a mutually-exclusive access is required: this calls for a mutex, semaphores, or any other lock system on the vehicles list, doesn't it?

Reply:
OpenTTD is not multithreaded so none of that is relevant.